xml/ha_cluster_lvm.xml

```diff
@@ -812,17 +821,63 @@ vdc                                   253:32   0   20G  0 disk
 │   └─cluster--vg2-test--lv           254:5    0   12G  0 lvm
 └─cluster--vg2-test--lv_mimage_1      254:4    0   12G  0 lvm
   └─cluster--vg2-test--lv             254:5    0   12G  0 lvm</screen>
-</example>
-<note>
- <title>Adjustments Needed</title>
- <para>
-  The following procedure is based on <xref linkend="ex.ha.clvm.setup.before"
-  xrefstyle="select:label"/>. Adjust the instructions to match your setup
-  and replace the names for the LVs, VGs, disks and the cluster MD device
-  accordingly.
- </para>
-</note>
-  <procedure>
+ </example>
+ 
+ <important>
+  <title>Avoiding Migration Failures</title>
+  <para>
+   Before you start the migration procedure, check the capacity and degree
+   of utilization of your logical and physical volumes. If the logical volume
+   uses 100% of the physical volume capacity, the migration might fail with an
+   <literal>insufficient free space</literal> error on the target volume.
+   How to prevent this migration failure, depends on the options used for
+   mirror log:
+  </para>
+  <qandaset>
+   <qandaentry>
+    <question>
+     <para>
+      Is the mirror log written to a different device (<option>disk</option>
+      option) or kept in memory (<option>core</option> option)?
+     </para>
+    </question>
+    <answer>
+     <para>
+      Before starting the migration, either enlarge the size of the physical
+      volume or reduce the size of the logical volume (to free more space for
+      the physical volume).
+     </para>
+    </answer>
+   </qandaentry>
+   <qandaentry>
+    <question>
+     <para>
+      Is the mirror log itself mirrored (<option>mirrored</option> option)
+      and allocated on the same device as the mirror leg?
+     </para>
+    </question>
+    <answer>
+     <para>
+      During migration, reduce unused padding space from the cluster MD array
+      by running the <command>mdadm</command> command with
+      <option>--data-offset</option>. See <xref linkend="step.data-offset"/>
+      for details. By default, <command>mdadm</command> reserves 16 MB space at
+      the start of a device, which can be used for reshaping (re-striping) the
+      array. Reducing this space allows to fit in the LVM data and metadata.
```

Comment:
Just to let you know one detail.

The 16MB is probably the superblock, right?

Although you mentioned "by default" and "at the start of the device", it may be a bit dangerous. The `mdadm` tool understands several metadata versions (0.90, 1.0, 1.1, 1.2, ...). According to the [mdadm manpage (option --metadata)](https://linux.die.net/man/8/mdadm):

>  The different sub-versions store the superblock at different locations on the device, either
>  at the end (for 1.0), at the start (for 1.1) or 4K from the start (for 1.2).  "1" is equivalent to
>  "1.2" (the commonly preferred 1.x format).  "default" is equivalent to "1.2".

If you refer really to the superblock, I would mention that:

    By default, <command>mdadm</command> reserves 16 MB space for the superblock...

Does that make sense?

Reply:
> The 16MB is probably the superblock, right?
> 
According to the [mdadm manpage (option --metadata)](https://linux.die.net/man/8/mdadm):
> 
> > The different sub-versions store the superblock at different locations on the device, either at the end (for 1.0), at the start (for 1.1) or 4K from the start (for 1.2).  "1" is equivalent to "1.2" (the commonly preferred 1.x format).  "default" is equivalent to "1.2".
> 
Very good question, I don't know (at least 'superblock' was not mentioned anywhere in the bug report). Handing over to @zzhou1 for this question.